### PR TITLE
Add kate rows RPC

### PR DESCRIPTION
**Pull request by aterentic-ethernal**
This PR adds `queryRows` RPC to support fetching and pushing rows to DHT on light clients. Since new RPC is added, node version is increased from 1.4.0 to 1.5.0. Dependency to avail-core will be updated once corresponding PR is merged.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -25,9 +25,9 @@ frame-system-rpc-runtime-api = { path = "pallets/system/rpc/runtime-api" }
 frame-system-benchmarking = { path = "pallets/system/benchmarking" }
 
 # Polygon Primitives
-da-primitives = { git="https://github.com/maticnetwork/avail-core", version = "0.4.2", tag = "da-primitives/v0.4.2" }
-kate = { git="https://github.com/maticnetwork/avail-core", version = "0.5.3", tag = "da-primitives/v0.4.2" }
-kate-recovery = { git="https://github.com/maticnetwork/avail-core", version = "0.7.1", tag = "da-primitives/v0.4.2" }
+da-primitives = { git="https://github.com/maticnetwork/avail-core", version = "0.4.2", branch = "AV-120-rows-in-dht" }
```
Comment (MiguelDD1) [inline Cargo.toml:28]: Deploy and tag `avail-core` changes before merge this.

File: rpc/kate-rpc/src/lib.rs [hunk 2]
```diff
@@ -141,6 +148,77 @@ where
...
+				block_length.chunk_size(),
+				&xts_by_id,
+				seed,
+			)
+			.map_err(|e| internal_err!("Flatten and pad block failed: {:?}", e))?;
+
+			let data = kate::com::par_extend_data_matrix(block_dims, &block)
+				.map_err(|e| internal_err!("Matrix cannot be extended: {:?}", e))?;
+
+			block_ext_cache.put(block_hash, (data, block_dims));
```
Comment (MiguelDD1) [inline rpc/kate-rpc/src/lib.rs:207]: QQ: Can we insert block in cache if and only if that block is finalized?

File: rpc/kate-rpc/src/lib.rs [hunk 3]
```diff
@@ -141,6 +148,77 @@ where
...
+			.map_err(|e| internal_err!("Invalid block number: {:?}", e))?
+			.ok_or_else(|| internal_err!("Missing block {}", block_id))?;
+		let block_hash = signed_block.block.header().hash();
+
+		let mut block_ext_cache = self
+			.block_ext_cache
+			.write()
+			.map_err(|_| internal_err!("Block cache lock is poisoned .qed"))?;
+
+		let block_length: BlockLength = self
```
Comment (MiguelDD1) [inline rpc/kate-rpc/src/lib.rs:170]: `block_length` only used inside `if`-block. Pls move it.

File: Cargo.toml (hunk 1 above)
Comment (aterentic-ethernal) [reply Cargo.toml:28]: Sure, I have mentioned that in description, but didn't want to postpone review.

File: rpc/kate-rpc/src/lib.rs (hunk 3 above)
Comment (aterentic-ethernal) [reply rpc/kate-rpc/src/lib.rs]: This affects three RPCs, I didn't wanted to spend much there, but RPCs are almost the same and code should be reused at some point.

File: rpc/kate-rpc/src/lib.rs (hunk 2 above)
Comment (aterentic-ethernal) [reply rpc/kate-rpc/src/lib.rs:207]: How to check if block is finalized?

File: Cargo.toml (hunk 1 above)
Comment (aterentic-ethernal) [reply Cargo.toml]: Done, but its not possible to use different tags for different crates from same repository, I don't know whats going on but we should revisit this at some point.

**Review by MiguelDD1 — APPROVED** [review]

**Pull request merged**